dask/dataframe/core.py

```diff
@@ -1175,11 +1180,12 @@ class DataFrame(_Frame):
     _partition_type = pd.DataFrame
     _token_prefix = 'dataframe-'
 
-    def __new__(cls, dask, name, columns, divisions):
+    def __new__(cls, dask, name, metadata, divisions):
         result = object.__new__(cls)
         result.dask = dask
         result._name = name
-        result.columns = tuple(columns)
+        result.metadata = metadata
+        result.columns = tuple(result.metadata.columns)
```

Comment:
This sort of thing should be a property.  https://en.wikipedia.org/wiki/Single_Source_of_Truth